app/components/primer/alpha/overlay.html.erb

```diff
@@ -0,0 +1,30 @@
+<%= show_trigger_button %>
```

Comment:
Are we planning to abstract this `show_trigger_button` and `role="dialog"` into a separate Modal component?

Reply:
I want to make `role` a prop instead of default to `dialog`.. and I think you even told me how and I must have not made the change here. So it should probably be an empty prop in the declaration?

I'm torn on the trigger being part of this component. Do you have an opinion? Most parent components that use Overlay will need a standard trigger, so if I remove it here it would just be the job of Dialog to supply the optional trigger.